lib/partials.ts

```diff
@@ -248,5 +248,7 @@ export const buildTemplate = async (
 		context.toJSON().children,
 	);
 
-	return stripExtraBlankLines(await hb.compile(template)(data));
+	return stripExtraBlankLines(
+		await Promise.resolve(hb.compile(template)(data)),
```

Comment:
Same here:
```suggestion
		hb.compile(template)(data),
```

Reply:
Because of the use `promiseHandlebars` here 
https://github.com/balena-io/contrato/blob/b2e71c992182a8cf3bd9e0d3c1bfa3025e63b604/lib/partials.ts#L33 
the compiler doesn't know that `hb.compile` can be `async`. 

I think just returning would be equivalent to returning without awaiting the promise, which is probably not what we want. That's why I added the `Promise.resolve`

Reply:
See the reply on https://github.com/balena-io/contrato/pull/79#discussion_r1910583954

Reply:
If the typings are improved now then this should change back to
```suggestion
		await hb.compile(template)(data),
```

Reply:
Done